features/features.go

```diff
@@ -161,18 +163,40 @@ var featuresSupportedCallbacks = map[SnapdFeature]func() (bool, string){
 	},
 	// UserDaemons requires user units
 	UserDaemons: func() (bool, string) {
-		if !release.SystemctlSupportsUserUnits() {
+		if !releaseSystemctlSupportsUserUnits() {
 			return false, "user session daemons are not supported on this system's distribution version"
 		}
 		return true, ""
 	},
-	// AppArmorPrompting requires a newer version of snapd with all the
+	// AppArmorPrompting requires AppArmor parser and kernel support for
+	// prompting, as well as a newer version of snapd with all the
 	// prompting components in place. TODO: change this callback once ready.
 	AppArmorPrompting: func() (bool, string) {
+		kernelFeatures, err := apparmorKernelFeatures()
+		if err != nil {
+			return false, fmt.Sprintf("error checking apparmor kernel features: %v", err)
+		}
+		if !strutil.ListContains(kernelFeatures, "policy:permstable32:prompt") {
+			return false, "apparmor kernel features do not support prompting"
```

Comment:
Nitpick: `the kernel does not support apparmor prompting`

The message here is really different than the one below for the parser. Make them human readable.